### ISSUE TITLE
Expose origin attribute in PortalActivateEvent

### ISSUE THREAD

**Issue by shhnjk** [WICG/portals#162]
I can imagine a future where a site is expecting a data in `PortalActivateEvent` from trusted site. 
But `PortalActivateEvent` can be triggered by any site (similar to `message` event), therefore origin attribute has to be exposed in `PortalActivateEvent`. Otherwise it'll be an XSS nightmare 😊

**Comment by jeremyroman** [comment]
Interesting suggestion. I think that the existing API surface is sufficient to allow authors to figure out what's going on here, but plumbing the origin here seems like it would be reasonably straightforward and be much easier to use for authors.

@a4sriniv, you looked at messaging APIs; WDYT of this?

**Comment by a4sriniv** [comment]
I think this is a good idea. I think of the data field in the portalactivate event as an author friendly alternative to just using postMessage before activation, so it makes sense to me to mirror the origin aspect of the message event (and it shouldn't be difficult to implement). 

But on a related note, we also have targetOrigin specified as part of the postMessage API parameters, which checks the origin of the destination frame before delivering the message. Do we need something similar for activate?

**Comment by jeremyroman** [comment]
> But on a related note, we also have targetOrigin specified as part of the postMessage API parameters, which checks the origin of the destination frame before delivering the message. Do we need something similar for activate?

An interesting thought. What would the semantics of that be if the guest contents navigates? Also, should it only limit delivery of the activation data, or should it block activation altogether?

**Comment by shhnjk** [comment]
targetOrigin Is also good to add :)

**Comment by a4sriniv** [comment]
> 
> An interesting thought. What would the semantics of that be if the guest contents navigates? Also, should it only limit delivery of the activation data, or should it block activation altogether?

My initial thought here was to fail the activation (resolve the promise with an error) as a guard against the guest contents navigating without the host's knowledge.

**Comment by domenic** [comment]
Is this still a good idea, given our desire to limit cross-origin communication?

**Comment by a4sriniv** [comment]
Yeah I don't think this applies anymore. I think we're going to omit portalactivatedata from the activate event for cross-origin portal activations.

**Comment by domenic** [comment]
Sounds good. Let's close this for now. I filed https://github.com/WICG/portals/issues/196 to track speccing those restrictions.

**Comment by jakearchibald** [comment]
Are we dropping adoption in the activate event? Otherwise, it seems like the page should know the origin of the page it's adopting.

**Comment by a4sriniv** [comment]
Ahh, I remembered this issue as using targetOrigin to limit which origins get activate data, I forgot about actually using the targetOrigin to stop activation altogether.

**Comment by domenic** [comment]
I'm a bit confused where this landed. I think there are several proposals being floated:

- Add a `targetOrigin` parameter to `activate()`, such that if the portal is currently navigated to something that doesn't match `targetOrigin`, then `activate()` fails.
- Add an `origin` member to `PortalActivateEvent`. This lets the portaled page know who is activating them. For example, it could let them better decide whether to call `adoptPredecessor()`, or how to interpret the data in `event.data` (e.g. "if activated by Aggregator X, then the data will probably be in Aggregator X format").
  - Since we are banning sending data in cross-origin activation, and potentially dropping `adoptPredecessor()`, this seems less useful. But I could still imagine use cases, for example displaying a "back to aggregator" link.
  - And, I just like the symmetry with `MessageEvent`.
  - But perhaps this is a cross-origin data leak?

Are either of these good ideas? Both of them?

**Comment by jeremyroman** [comment]
We could defer adding this to later if we wanted to, though I suspect it will eventually be useful especially in combination with `adoptPredecessor`.

There might be a world in which we become worried about sites setting using their hostname to convey information (though I'm not convinced that is convenient enough that it would actually be in widespread use), but that will arise elsewhere in the platform (location.ancestorOrigins etc) and mitigations exist (like heuristically or unconditionally censoring to eTLD+1).
